### PR TITLE
[None][doc] Add sparse attention tech blog

### PR DESCRIPTION
**Pull request by heyuhhh**
## Release Notes

* **Documentation**
  * Added a comprehensive technical blog post documenting TensorRT-LLM's sparse attention framework, including design philosophy, architecture overview, integrated algorithms, and memory management strategies.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by juney-nvidia — APPROVED** [review]
LGTM

File: docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+## Overview of Sparse Attention in TensorRT LLM
+
+A key challenge in deploying sparse attention at scale is the diversity of existing methods: they differ in prediction strategy (heuristic vs. learned), sparsity granularity (page-level vs. token-level), integration point (framework-driven vs. kernel-internal), and target attention architecture (MQA/MHA/GQA vs. MLA). Implementing each algorithm as an isolated feature leads to fragmented codepaths, duplicated infrastructure, and high maintenance costs. To address this, TensorRT-LLM introduces a **unified, extensible sparse attention framework** that factors out the common structure shared across algorithms—prediction hooks, standardized sparse indices, attention operator interfaces, and auxiliary memory management—into a reusable pipeline. Under this design, integrating a new sparse attention algorithm primarily reduces to implementing its prediction logic and producing indices in the expected format; the framework handles KV cache layout, kernel dispatch, and memory management transparently.
```
Comment (bobboli) [inline docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md]: Overall, this paragraph is describing prediction-based sparse attention.
My feeling is that some part, like Sparse Attention Config, is universal.
The prediction-based framework support is mainly demonstrated by RocketKV. (I am not sure how much that DSA relies on that, but Skip Softmax Attention almost only uses the Sparse Attention Config).
Better to clearly indicate these differences.

File: docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+This blog focuses on the **framework-level** design that is common across algorithms. For kernel-level optimizations (Skip Softmax Attention, sparse MLA kernel), please refer to the dedicated documents linked in the algorithm sections below.
+
+## Sparse Attention Framework Design
+
+The sparse attention framework in TensorRT LLM is designed to provide a common runway for different algorithms while hiding most of the complexity from end users. Our goal is to make it straightforward for developers to integrate new sparse attention methods through a set of unified interfaces, without modifying the core attention kernels or the serving infrastructure.
+
+### Design Philosophy
+
+Despite their diversity, most sparse attention algorithms follow a common pattern:
```
Comment (bobboli) [inline docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md]: most sparse attention algorithms  -> prediction based sparse attention algorithms.

**Review by bobboli — APPROVED** [review]

File: docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+
+Because the prediction step can be computationally heavy—especially in low-latency scenarios—it is typically implemented with custom Triton or CUDA kernels rather than generic PyTorch ops.
+
+### Attention Operator Design
+
+From an operator perspective, we recommend integrating sparse attention through the TRTLLM attention backend (`TrtllmAttention`). Under this backend, sparse attention is realized inside a common attention operator: the prediction module produces indices, and `AttentionOp` turns them into concrete KV cache layouts and attention workloads. Figure 2 shows how these pieces work together.
+
+<div align="center">
+<figure>
+  <img src="https://github.com/heyuhhh/TensorRT-LLM/blob/user/yuhangh/add_sprase_attention_tech_blog/docs/source/blogs/media/tech_blog17_sparse_attention_op.png" width="700">
```
[the remark this answers is not in the document]
Comment (heyuhhh) [reply docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md]: Updated

File: docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+
+## Sparse Attention Framework Design
+
+The sparse attention framework in TensorRT LLM is designed to provide a common runway for different algorithms while hiding most of the complexity from end users. Our goal is to make it straightforward for developers to integrate new sparse attention methods through a set of unified interfaces, without modifying the core attention kernels or the serving infrastructure.
+
+### Design Philosophy
+
+Despite their diversity, most sparse attention algorithms follow a common pattern:
+
+- **Predict** which tokens or blocks are important.
```
[the remark this answers is not in the document]
Comment (heyuhhh) [reply docs/source/blogs/tech_blog/blog17_Sparse_Attention_in_TensorRT-LLM.md]: Done
